cl/search/management/commands/sweep_indexer.py

```diff
@@ -0,0 +1,392 @@
+
+    :param es_document: The ES document class type.
+    :param instance_id: The DB instance ID related to the ES document.
+    :return: A two-tuple containing the Elasticsearch document ID and the
+    parent ID.
+    """
+
+    if es_document is ESRECAPDocument:
+        doc_id = ES_CHILD_ID(instance_id).RECAP
+    elif es_document is OpinionDocument:
+        doc_id = ES_CHILD_ID(instance_id).OPINION
+    else:
+        doc_id = instance_id
+    return doc_id
+
+
+class Command(VerboseCommand):
+    help = "Sweep indexer for Elasticsearch documents."
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.options = {}
+        self.sweep_indexer_action = settings.ELASTICSEARCH_SWEEP_INDEXER_ACTION
+        self.queue = settings.ELASTICSEARCH_SWEEP_INDEXER_QUEUE
+        self.chunk_size = settings.ELASTICSEARCH_SWEEP_INDEXER_CHUNK_SIZE
+        self.poll_interval = settings.ELASTICSEARCH_SWEEP_INDEXER_POLL_INTERVAL
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--testing-mode",
+            action="store_true",
+            help="Use this flag only when running the command in tests based on TestCase",
+        )
+
+    def handle(self, *args, **options):
+        super().handle(*args, **options)
+        self.options = options
+
+        testing_mode = self.options.get("testing_mode", False)
+        while True:
+            self.stdout.write(f"Starting a new sweep indexer cycle.")
+            self.execute_sweep_indexer_cycle()
+            document_counts = get_documents_processed_count_and_restart()
+            self.stdout.write(
+                f"\rSweep indexer '{self.sweep_indexer_action}' cycle finished successfully."
+            )
+            logger.info(f"\rDocuments Indexed: {document_counts}")
+            if testing_mode:
+                # Only execute one cycle for testing purposes.
+                break
+
+    def execute_sweep_indexer_cycle(self) -> None:
+        """Executes a sweep indexing cycle by processing documents starting
+        from the last processed document, if it's stored, or starting from
+        scratch if the command initiates a new cycle.
+
+        :return: None
+        """
+
+        document_name, last_document_id = get_last_document_processed()
+        start_model_index = find_starting_model(
+            list_supported_models, document_name
+        )
+        if start_model_index is not None and last_document_id:
+            # create a sub-list from the start model to the end, then reverse
+            # it to process it as a stack.
+            models_stack = list_supported_models[start_model_index:][::-1]
+            self.process_documents(models_stack, last_document_id)
+        else:
+            # Copy and reverse list_supported_models to process it as a stack.
+            models_stack = list_supported_models[::-1]
+            self.process_documents(models_stack)
+
+    def process_documents(
```

Comment:
Maybe rename this as `process_model_stack`? It's a bit confusing to have `process_documents`, which calls `process_queryset`?

Reply:
sure this name is better, changed!